src/components/Tag/types.ts

```diff
@@ -0,0 +1,24 @@
+export type TagSizes = 's' | 'm';
+export type TagStyles = 'ghost' | 'outlined';
+
+export type NonMutuallyExclusiveProps = {
+  size?: TagSizes;
+  icon?: string;
+  emoji?: string;
+  color?: string;
+  style?: TagStyles;
+  disabled?: boolean;
+  clickable?: boolean;
+  removable?: boolean;
+  expandable?: boolean;
+};
+
+export type TagProps = NonMutuallyExclusiveProps &
+  (
+    | { emoji?: string; icon?: undefined; color?: undefined }
+    | { emoji?: undefined; icon?: string; color?: undefined }
+    | { emoji?: undefined; icon?: undefined; color?: string }
+  ) & (
+    | { removable?: boolean, expandable?: undefined }
+    | { removable?: undefined, expandable?: boolean }
+  )
```

Comment:
Just a small cosmetic suggestion, makes it more readable. Feel free to ignore it

```ts
export type BaseTagProps = {
  size?: TagSizes;
  style?: TagStyles;
  disabled?: boolean;
  clickable?: boolean;
};

export type EmojiTagProps = BaseTagProps & {
  emoji: string;
  icon?: never;
  color?: never;
  removable?: boolean;
  expandable?: never;
};

export type IconTagProps = BaseTagProps & {
  emoji?: never;
  icon: string;
  color?: never;
  removable?: never;
  expandable?: boolean;
};

export type ColorTagProps = BaseTagProps & {
  emoji?: never;
  icon?: never;
  color: string;
  removable?: boolean;
  expandable?: boolean;
};

export type TagProps = EmojiTagProps | IconTagProps | ColorTagProps;
```